src/tests/VestedRewardsDistribution-integration.t.sol

```diff
@@ -0,0 +1,441 @@
+        GodMode.setWard(address(vestedRewardsDist1), address(this), 1);
+        GodMode.setWard(address(vestedRewardsDist2), address(this), 1);
+
+        GodMode.setWard(vestedRewardsDist1.dssVest(), address(this), 1);
+        GodMode.setWard(vestedRewardsDist2.dssVest(), address(this), 1);
+    }
+
+    function test_add_rewards_distribution() public {
+        address rewardsDist = address(0); //test address
+        vm.expectEmit(true, false, false, true);
+        emit Set(rewardsDist, RANDOM_INTERVAL);
+        job.set(rewardsDist, RANDOM_INTERVAL);
+        assertTrue(job.has(rewardsDist));
+        assertEq(job.intervals(rewardsDist), RANDOM_INTERVAL);
+    }
+
+    function test_add_rewards_distribution_revert_auth() public {
+        vm.prank(address(1));
+        vm.expectRevert("VestedRewardsDistributionJob/not-authorized");
+        job.set(address(0), RANDOM_INTERVAL);
+    }
+
+    function test_add_rewards_distribution_overwrite_duplicate() public {
+        address rewardsDist = address(0); //test address
+        job.set(rewardsDist, RANDOM_INTERVAL);
+        job.set(rewardsDist, RANDOM_INTERVAL + 1);
+
+        assertEq(job.intervals(rewardsDist), RANDOM_INTERVAL + 1);
+    }
+
+    function test_remove_rewards_distribution() public {
+        address rewardsDist = address(0); //test address
+        job.set(rewardsDist, RANDOM_INTERVAL);
+        vm.expectEmit(true, false, false, false);
+        emit Rem(rewardsDist);
+        job.rem(rewardsDist);
+        assertFalse(job.has(rewardsDist));
+        assertEq(job.intervals(rewardsDist), 0);
+    }
+
+    function test_remove_rewards_distribution_revert_auth() public {
+        vm.prank(address(1));
+        vm.expectRevert("VestedRewardsDistributionJob/not-authorized");
+        job.rem(address(0));
+    }
+
+    function test_remove_rewards_distribution_revert_not_found() public {
+        address rewardsDist = address(0); //test address
+        vm.expectRevert(abi.encodeWithSelector(VestedRewardsDistributionJob.NotFound.selector, rewardsDist));
+        job.rem(rewardsDist);
+    }
+
+    function test_modify_distribution_interval() public {
+        uint256 newInterval = RANDOM_INTERVAL + 1;
+        vm.expectEmit(true, false, false, true);
+        emit Set(address(vestedRewardsDist1), newInterval);
+        job.set(address(vestedRewardsDist1), newInterval);
+        assertEq(job.intervals(address(vestedRewardsDist1)), newInterval);
+    }
+
+    function test_modify_distribution_interval_revert_auth() public {
+        vm.prank(address(1));
+        vm.expectRevert("VestedRewardsDistributionJob/not-authorized");
+        job.set(address(vestedRewardsDist1), RANDOM_INTERVAL);
+    }
+
+    function test_modify_distribution_interval_revert_invalid_arg() public {
+        vm.expectRevert(abi.encodeWithSelector(VestedRewardsDistributionJob.InvalidInterval.selector));
+        job.set(address(vestedRewardsDist1), 0);
+    }
+
+    function test_work() public {
```

Comment:
I think since `work()` is no longer a view function it should be also tested to prove that it works as expected i.e. it calls `work()` and it has the desired side effects.

Reply:
Do you mean `workable`?

Reply:
I meant `workable()`!

Reply:
If you look closely, there are some assertions through `test_work` and others that test the output of `workable`.
I tried to split them into separate test cases, but it just gets more verbose for the same effect.